docs_src/manuals/user_manual/execution_model.hrst

```diff
@@ -31,15 +31,113 @@ execution policy. A computation combines several multistages and executes one mu
   previous stages of the multistage have not already been applied to the indices in the same column with smaller ``k``.
 
 
+----------------------------------
+Extended Compute Domain of a Stage
+----------------------------------
+
+For a clear specification of access restrictions, we need to introduce the concept of extended compute domains.
+
+If a stage is accessing neighbor grid points of field `a`, the computation will consume halo points at the boundary of the compute domain.
+In case field `a` was written in a previous stage of the same multi-stage, the compute domain of the first stage needs to be extended, such
+that the *extended compute domain* of the stage covers the extent of the read access in the later stage.
+
 -------------------
 Access restrictions
 -------------------
 
-The execution model imposes restrictions on how accessors can be evaluated. The following restrictions apply:
+The execution model imposes restrictions on how accessors can be evaluated. These restrictions are not checked by GridTools,
+violating these rules results in undefined behavior.
+
+Definitions:
+
+ - *Writing/reading to/from an accessor* is short for *writing/reading to/from a field through an accessor*.
+   Note that the same field could be bound to different accessors, though this is not recommended.
+
+- *Previous/later statement* is short for *previous/later statement in the same stage or any previous/later stage*.
+
+
+The following rules apply to accesses in stages *within the same multi-stage*:
+
+ 1. A stage may read from an accessor unconditionally, if the same accessor is never written to.
+
+ 2. A stage may write to an accessor only once.
```

Comment:
This is at least unclear, but maybe also incorrect. There is no problem with a write-write dependency within a stage. It would be a problem, if there is a write-read-write dependency, but this is not what this sentence is about. I also wonder in this sentence, if this would forbid that we write to an accessor once within the if, and once within the else clause (which is perfectly fine).

I think you probably want to say what we discussed yesterday, that for each field, there must be one and only one stage where the field is bound to one (or more) inout_accessors, otherwise the extent computation gets in trouble. But maybe that could be a limitation that we describe separately, because we can avoid it with make_stage_with_extents.